Manage theme fonts

matiasbenedetto opened the pull request:
## What?
This PR consolidates the previous work around handling theme fonts using the plugin. 
The new changes:
- Allows previewing theme font families and font faces.
- Allows removing font families.
- Allows removing font faces.
- Removes the "embed google font" and "embed local font" options from the dashboard menu and replaces it for "Manage fonts".

## Why?
Because currently is possible just add fonts but not remove them.


## Screencast
[Screencast from 18-10-22 00:31:11.webm](https://user-images.githubusercontent.com/1310626/196302664-2a55ad6c-f9f7-43c1-8db9-6baf73e0d572.webm)

matiasbenedetto committed:
Starting the implementation of the manage fonts functionality

matiasbenedetto committed:
Implemting client side font removal

matiasbenedetto committed:
sending font changes from client to server

matiasbenedetto committed:
refining implementation

matiasbenedetto committed:
Adding string translations in JS files

matiasbenedetto committed:
Make strings translatable in JS files

matiasbenedetto committed:
adding slide up and down for the font families component

matiasbenedetto committed:
consolidate "add  google fonts" and "add local font" optios into "Manage Fonts" menu item

matiasbenedetto committed:
Add the table head of fontfamily clickeable to expland or collapse the family

matiasbenedetto committed:
Add manage fonts link to embed font success notification

matiasbenedetto committed:
checking the right permissions to handle the  font changes

matiasbenedetto committed:
renaming php file to reflect the current functionality

matiasbenedetto committed:
avoids collapsing the family component when the remove family is pressed

jffng commented on admin/class-manage-fonts.php:
```diff
@@ -53,6 +58,51 @@ function can_read_and_write_font_assets_directory () {
+        <a href="<?php echo admin_url( 'themes.php?page=add-google-font-to-theme-json' ); ?>" class="page-title-action"><?php _e('Add Google Font', 'create-block-theme'); ?></a>
+        <a href="<?php echo admin_url( 'themes.php?page=add-local-font-to-theme-json' ); ?>" class="page-title-action"><?php _e('Add Local Font', 'create-block-theme'); ?></a>
+        <hr class="wp-header-end" />
+        <p><?php printf( esc_html__( 'These are the fonts currently available in your theme (%1$s).', 'create-block-theme' ), esc_html( $theme_name ) ); ?></p>
```
If the theme itself bundles a custom font, should it display them? For example, I tested with Pixl which bundles DMMono, but I do not see it: 

<img width="1624" alt="Screen Shot 2022-10-18 at 12 32 57 PM" src="https://user-images.githubusercontent.com/5375500/196418416-cc575aa6-26ea-4e0e-aac3-b9bb0fd503d0.png">

matiasbenedetto replied:
Yep, all the fonts present in theme.json should be listed. This was a glitch happening with some themes. It should be solved with the latest changes.

madhusudhand replied:
I see it working now!

jffng reviewed (commented):
This is great! :clap:

I loaded a custom google font, and applied it in the site editor > global styles > typography, but it is not rendering: 

<img width="1624" alt="Screen Shot 2022-10-18 at 12 34 58 PM" src="https://user-images.githubusercontent.com/5375500/196418855-43930522-e529-47fd-ab35-af21faf9456e.png">

The front-end and post editor are working.

matiasbenedetto committed:
renaming class

matiasbenedetto committed:
load font faces when Gutenberg is not activated

matiasbenedetto committed:
avoid problem parsing JSON from input field

matiasbenedetto committed:
replacing deprecated IconButton component by Button

matiasbenedetto committed:
fixing mime type check

matiasbenedetto committed:
Fix font preview styles

matiasbenedetto committed:
avoid showing modal with undefined data

matiasbenedetto committed:
handling font families without font faces defined

matiasbenedetto committed:
adding modal with info

matiasbenedetto commented:
> I loaded a custom google font, and applied it in the site editor > global styles > typography, but it is not rendering:

I think this was caused by a glitch that was making the fonts not being saved and because of that not loaded in the editor. Please, try again with the latest changes.

madhusudhand commented on package.json:
```diff
@@ -0,0 +1,25 @@
+{
+    "name": "05-recipe-card-esnext",
```
update the fields to the appropriate values.

matiasbenedetto replied:
updated!

madhusudhand reviewed (commented):
Great work @matiasbenedetto 🎉  It is much simplified and working great.

madhusudhand commented:
Can you pick `name` and show it. And if it doesn't exists then fallback to `familyName`

<img width="327" alt="image" src="https://user-images.githubusercontent.com/1935113/196981885-0bea6ac0-7a02-4412-87ae-62119c62e650.png">

matiasbenedetto committed:
default font family title to  fontFamily name and fallback to  fontFamily family

matiasbenedetto committed:
updating package.json data

matiasbenedetto committed:
version bump

matiasbenedetto committed:
version bump corrrection and update of README

matiasbenedetto committed:
Adding updated screenshots

matiasbenedetto commented:
> Can you pick name and show it. And if it doesn't exists then fallback to familyName

Done!

matiasbenedetto committed:
Add the ability to select/unselect all google font variants

matiasbenedetto committed:
uncheck select all variant checkbox when other font is selected

matiasbenedetto committed:
avoid error when there is no a font selected.

Previously we were  trying to iterate over variant even when no font was selected in the select input

matiasbenedetto committed:
Enable/disable submit button reacting to select/unselect all variants

matiasbenedetto committed:
fix select/unselect all variants functionality to send to server the right data

jffng commented on src/font-family.js:
```diff
@@ -0,0 +1,81 @@
+}
+
+FontFamily.defaultProps = {
+    demoText: __("The quick brown fox jumps over the lazy dog."),
```
```suggestion
    demoText: __("The quick brown fox jumps over the lazy dog.", "create-block-theme"),
```

jffng commented on src/font-family.js:
```diff
@@ -0,0 +1,81 @@
+                                    <td>{__('Style')}</td>
+                                    <td>{__('Weight')}</td>
+                                    <td>{__('Preview')}</td>
+                                    {/* <td>{__('Edit')}</td> */}
```
```suggestion
```

jffng commented on src/font-face.js:
```diff
@@ -0,0 +1,27 @@
+            <td>{fontFace.fontStyle}</td>
+            <td>{fontFace.fontWeight}</td>
+            <td className="demo-cell"><p style={ demoStyles }>{demoText}</p></td>
+            {/* <td><Button variant="secondary">Edit</Button></td> */}
```
```suggestion
```

jffng commented on build/index.asset.php:
```diff
@@ -0,0 +1 @@
+<?php return array('dependencies' => array('react', 'wp-components', 'wp-element', 'wp-polyfill'), 'version' => 'adf49c3b45f9ee2c8230');
```
Now that there is a compiled code for the plugin, is it common to include the build inside the repo, or should that be done another way?

matiasbenedetto replied:
Good catch, thanks! I will remove the build files from this PR. We will need to add extra steps in the release process to build the app. I will follow up on that task in another PR.

matiasbenedetto replied:
The follow up PR: https://github.com/WordPress/create-block-theme/pull/131

jffng reviewed (commented):
Looking good and working well for me, a few nit picks.

matiasbenedetto committed:
remove not needed comment

Co-authored-by: Jeff Ong <jonger4@gmail.com>

matiasbenedetto committed:
remove not needed comment

Co-authored-by: Jeff Ong <jonger4@gmail.com>

matiasbenedetto committed:
adding translation domain

Co-authored-by: Jeff Ong <jonger4@gmail.com>

matiasbenedetto committed:
Merge pull request #128 from WordPress/add/select-all-google-font-variants

Add the ability to select/unselect all google font variants

matiasbenedetto committed:
removing build folder from repo

matiasbenedetto committed:
Merge branch 'trunk' into try/manage-fonts

matiasbenedetto committed:
Remove version changes because we will handle this automatically

matiasbenedetto committed:
Merge branch 'trunk' into try/manage-fonts

jffng reviewed (approved):
Looking and working great, pre-approving with the message that we should add something to the readme now that the plugin requires a build step for development.

matiasbenedetto commented:
> Looking and working great, pre-approving with the message that we should add something to the readme now that the plugin requires a build step for development.

Ok, I think we could add a section around contributing/developing of the plugin. I'm adding an issue as a follow-up: https://github.com/WordPress/create-block-theme/issues/132
